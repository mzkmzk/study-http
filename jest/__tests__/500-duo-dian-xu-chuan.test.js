const curl = require('../curl/index')

// 内容为: 123456789ABCDEF
/**
 * 考点: 分段请求资源, 例如`Range: bytes=1-3`这种形式请求资源
 */
it('多点续传 训练1', async () => {
  const { status, body, row } = await curl(`curl -vs -H 'Range: bytes=1-3' -H 'Host: study-http.404mzk.com' http://47.115.81.18:9500/test.log`)

  expect(status).toBe(206)
  expect(body).toBe('234')
  expect(row).toMatchSnapshot()
})

/**
 * 考点: 分段请求资源, `10-`表示从第11个字节开始获取之后的内容
 */
it('多点续传 训练2', async () => {
  const { status, body, row} = await curl(`curl -vs -H 'Range: bytes=10-' -H 'Host: study-http.404mzk.com' http://47.115.81.18:9500/test.log`)

  expect(status).toBe(206)
  expect(body).toBe('BCDEF')
  expect(row).toMatchSnapshot()
})

/**
 * 考点: 分段请求资源, `10-`表示从第11个字节开始获取之后的内容
 */
it('多点续传 训练3', async () => {
  const { status, body, row } = await curl(`curl -vs -H 'Range: bytes=-3' -H 'Host: study-http.404mzk.com' http://47.115.81.18:9500/test.log`)

  expect(status).toBe(206)
  expect(body).toBe('DEF')
  expect(row).toMatchSnapshot()
})

/**
 * 考点: 多段传输的内容返回, 和最后一个boundary以--结束
 */
it('多点续传 训练4', async () => {
  const { status, body, row } = await curl(`curl -vs -H 'Range: bytes=0-0,10-11' -H 'Host: study-http.404mzk.com' http://47.115.81.18:9500/test.log`)

  expect(status).toBe(206)
  //expect(body).toBe('234')
  //思考 考点为内容填写和
  expect(row).toMatchSnapshot()
})

/**
 * 考点: 下载过程中以If-Match确认片段包是否仍有效
 */
it('多点续传 训练5', async () => {
  const { status, body, row} = await curl(`curl -vs -H 'Range: bytes=0-0' -H 'If-Match: "error-etag"' -H 'Host: study-http.404mzk.com' http://47.115.81.18:9500/test.log`)

  expect(status).toBe(412)
  //expect(body).toBe('234')
  expect(row).toMatchSnapshot()
})


