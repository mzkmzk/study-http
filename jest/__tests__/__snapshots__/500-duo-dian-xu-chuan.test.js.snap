// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`多点续传 训练1 1`] = `
"GET /test.log HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Range: bytes=1-3
Host: study-http.404mzk.com

HTTP/1.1 206 Partial Content
Server: nginx/1.17.10
Date: mock-time
Content-Type: application/octet-stream
Content-Length: 3
Last-Modified: mock-time
Connection: keep-alive
ETag: \\"5ed8f9a5-f\\"
Content-Range: bytes 1-3/15

234"
`;

exports[`多点续传 训练2 1`] = `
"GET /test.log HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Range: bytes=10-
Host: study-http.404mzk.com

HTTP/1.1 206 Partial Content
Server: nginx/1.17.10
Date: mock-time
Content-Type: application/octet-stream
Content-Length: 5
Last-Modified: mock-time
Connection: keep-alive
ETag: \\"5ed8f9a5-f\\"
Content-Range: bytes 10-14/15

BCDEF"
`;

exports[`多点续传 训练3 1`] = `
"GET /test.log HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Range: bytes=-3
Host: study-http.404mzk.com

HTTP/1.1 206 Partial Content
Server: nginx/1.17.10
Date: mock-time
Content-Type: application/octet-stream
Content-Length: 3
Last-Modified: mock-time
Connection: keep-alive
ETag: \\"5ed8f9a5-f\\"
Content-Range: bytes 12-14/15

DEF"
`;

exports[`多点续传 训练4 1`] = `
"GET /test.log HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Range: bytes=0-0,10-11
Host: study-http.404mzk.com

HTTP/1.1 206 Partial Content
Server: nginx/1.17.10
Date: mock-time
Content-Type: multipart/byteranges; boundary=mock-boundary-00001
Content-Length: 227
Last-Modified: mock-time
Connection: keep-alive
ETag: \\"5ed8f9a5-f\\"


--mock-boundary-00001
Content-Type: application/octet-stream
Content-Range: bytes 0-0/15

1
--mock-boundary-00001
Content-Type: application/octet-stream
Content-Range: bytes 10-11/15

BC
--mock-boundary-00001--
"
`;

exports[`多点续传 训练5 1`] = `
"GET /test.log HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Range: bytes=0-0
If-Match: \\"error-etag\\"
Host: study-http.404mzk.com

HTTP/1.1 412 Precondition Failed
Server: nginx/1.17.10
Date: mock-time
Content-Type: text/html
Content-Length: 174
Connection: keep-alive

<html>
<head><title>412 Precondition Failed</title></head>
<body>
<center><h1>412 Precondition Failed</h1></center>
<hr><center>nginx/1.17.10</center>
</body>
</html>
"
`;
